### DISCUSSION TITLE
Needs.job.outputs issue

### DISCUSSION THREAD

**Discussion by aleks-ivanov** [community/community#27198] [Actions]
<p>Share in my pain, please! :cry:</p>
<p>How can an output from one job, work in any  property OR inside the whole job, but NOT work in its <strong>if</strong> property…</p>
<pre><code class="lang-auto">cd:
    needs: ci
    if: ${{ needs.ci.outputs.new_version }} == '' # is ignored...
    runs-on: windows-latest
    env:
      NEW_VERSION: ${{ needs.ci.outputs.new_version }} # works
    steps:
      - name: Checkout repository
        id: checkout_repo
        uses: actions/checkout@v2 

      - run: echo ${{ needs.ci.outputs.new_version }} # works
</code></pre>
<p>Am I doing something wrong ?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="AleksIvanovSinglet" data-post="1" data-topic="149156">
<div class="title">
<div class="quote-controls"></div>
 AleksIvanovSinglet:</div>
<blockquote>
<pre><code class="lang-yaml">    if: ${{ needs.ci.outputs.new_version }} == '' # is ignored...
</code></pre>
</blockquote>
</aside>
<p>You need to either wrap the <em>whole</em> expression in <code>${{ ... }}</code>, or (only for <code>if</code>!) omit it as described in <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#about-contexts-and-expressions">About contexts and expressions</a>.</p>

**Comment by aleks-ivanov** [comment]
<p>Yeah… I just realized that about ten lines above, I am doing it properly and here my mind just blanked I guess… Thank you for waking me up a bit! :slight_smile:</p>
<p>it is late… :smiley:</p>
